src/test/java/org/wildfly/security/keystore/KeyStoreUtilTest.java

```diff
@@ -0,0 +1,256 @@
+        System.out.println("Testing JKS...");
+        Certificate jkscert = generateCertificate();
+        String filename = "testks.jks";
+        String alias = "alias";
+        char[] password = "password".toCharArray();
+
+        generateKeyStoreWithKey(filename, "jks", alias, password, jkscert);
+
+        KeyStore loadedStore = KeyStoreUtil.loadKeyStore(providerSupplier, null, new FileInputStream(new File(workingDir, filename)), filename, password);
+        Assert.assertNotNull(loadedStore);
+        Certificate loadedCert = loadedStore.getCertificate(alias);
+
+        Assert.assertEquals(jkscert, loadedCert);
+    }
+
+    @Test
+    public void testJCEKS() throws CertificateException, NoSuchAlgorithmException, KeyStoreException, IOException {
+        System.out.println("Testing JCEKS...");
+        Certificate jkscert = generateCertificate();
+        String filename = "testks.pkcs12";
+        String alias = "alias";
+        char[] password = "password".toCharArray();
+
+        generateKeyStoreWithKey(filename, "jceks", alias, password, jkscert);
+
+        KeyStore loadedStore = KeyStoreUtil.loadKeyStore(providerSupplier, null, new FileInputStream(new File(workingDir, filename)), filename, password);
+        Assert.assertNotNull(loadedStore);
+        Certificate loadedCert = loadedStore.getCertificate(alias);
+
+        Assert.assertEquals(jkscert, loadedCert);
+    }
+
+    @Test
+    public void testPKCS12() throws CertificateException, NoSuchAlgorithmException, KeyStoreException, IOException {
+        System.out.println("Testing PKCS12...");
+        Certificate jkscert = generateCertificate();
+        String filename = "testks.asdf";
+        String alias = "alias";
+        char[] password = "password".toCharArray();
+
+        generateKeyStoreWithKey(filename, "pkcs12", alias, password, jkscert);
+        generateKeyStoreWithKey(filename+"b", "pkcs12", alias, password, jkscert);
+        generateKeyStoreWithKey(filename+"c", "pkcs12", alias, password, jkscert);
+
+        KeyStore loadedStore = KeyStoreUtil.loadKeyStore(providerSupplier, null, new FileInputStream(new File(workingDir, filename)), filename, password);
+        Assert.assertNotNull(loadedStore);
+        Certificate loadedCert = loadedStore.getCertificate(alias);
+
+        Assert.assertEquals(jkscert, loadedCert);
+    }
+
+    @Test
+    public void testBKS() throws CertificateException, NoSuchAlgorithmException, KeyStoreException, IOException {
+        System.out.println("Testing BKS...");
+        Certificate jkscert = generateCertificate();
+        String filename = "testks.bks";
+        String alias = "alias";
+        char[] password = "password".toCharArray();
+        boolean bcfailed = false;
+        try {
+            generateKeyStoreWithKey(filename, "bks", alias, password, jkscert);
+        } catch (Exception e) {
+            bcfailed = true;
```

Comment:
There is already test dependency on bcmail-jdk15on. Would it be possible to add another BC test dependency - bcprov-jdk15on-160.jar to test BKS?

Reply:
Done.